### PR TITLE
feat: init from configuration file

### PR DESCRIPTION
**Pull request by Abingcbc**
Fix: https://github.com/casdoor/casdoor/issues/851

### CODE REVIEW

File: object/init.go [hunk 1]
```diff
@@ -221,3 +254,100 @@ func initBuiltInProvider() {
 	}
 	AddProvider(provider)
 }
+
+func readInitDataFromFile(filePath string) *InitData {
```
Comment (hsluoyz) [inline object/init.go:258]: Refer to this code in the util pkg:

```go
func readI18nFile(language string) *I18nData {
	s := util.ReadStringFromPath(getI18nFilePath(language))

	data := &I18nData{}
	err := util.JsonToStruct(s, data)
	if err != nil {
		panic(err)
	}
	return data
}
```

File: object/init.go [hunk 2]
```diff
@@ -221,3 +254,100 @@ func initBuiltInProvider() {
...
+func initDefinedLdap(ldap *Ldap) {
+	existedLdap := GetLdap(ldap.Id)
+	if existedLdap != nil {
+		return
+	}
+	AddLdap(ldap)
+}
+
+func initDefinedProvider(provider *Provider) {
+	existedProvider := GetProvider(provider.Owner + "/" + provider.Name)
```
Comment (hsluoyz) [inline object/init.go:348]: Use `provider.GetId()`

File: object/init.go [hunk 3]
```diff
@@ -15,11 +15,22 @@
...
 
 import (
+	"encoding/json"
 	"io/ioutil"
+	"os"
 
 	"github.com/casdoor/casdoor/util"
 )
 
+type InitData struct {
```
Comment (hsluoyz) [inline object/init.go:25]: Put all new code into new file: `init_data.go`

File: object/init.go [hunk 4]
```diff
@@ -221,3 +254,100 @@ func initBuiltInProvider() {
...
+	defer jsonFile.Close()
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+
+	var initData InitData
+	json.Unmarshal(byteValue, &initData)
+	return &initData
+}
+
+func initDefinedOrganization(organization *Organization) {
+	existedOrganization := getOrganization(organization.Owner, organization.Name)
```
Comment (hsluoyz) [inline object/init.go:276]: Change all to `existed` for simplicity

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: object/init.go (hunk 4 above)
Comment (Abingcbc) [reply object/init.go]: done

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**